.github/workflows/ci-template-check.yml

```diff
@@ -0,0 +1,16 @@
+# Checks if a PR makes any changes that ought to be shared via templating.
+# See the called workflow in the scitools/workflows repo for more details.
```

Comment:
Name can be a long string, including spaces, but this is inconsistent with the ones we already have. I am happy to change if preferred. In my demo it was:

```
Check if any changes need sharing via templating
```